rust/numaflow-core/src/source.rs

```diff
@@ -266,6 +280,12 @@ impl Source {
                     }
                 };
                 let n = messages.len();
+
+                // If no messages are read, continue to read the next batch.
+                if n == 0 {
+                    continue;
+                }
```

Comment:
will this end up as "missing metric" if the source does not have any data at all?

Reply:
yeah, it will end up in missing metric. I think its better publish metrics even when no messages are read. Reverting the change.